tasks.py

```diff
@@ -1,34 +1,93 @@
+import os
 from pathlib import Path
+from shutil import which
 
 from invoke import task
 
-PKG_NAME = 'more_categories'
-PKG_PATH = Path(f'pelican/plugins/{PKG_NAME}')
-FLAKE8 = 'flake8'
-ISORT = 'isort'
-PYTEST = 'pytest'
+PKG_NAME = "more_categories"
+PKG_PATH = Path(f"pelican/plugins/{PKG_NAME}")
+
+ACTIVE_VENV = os.environ.get("VIRTUAL_ENV", None)
+VENV_HOME = Path(os.environ.get("WORKON_HOME", "~/.local/share/virtualenvs"))
+VENV_PATH = Path(ACTIVE_VENV) if ACTIVE_VENV else (VENV_HOME / PKG_NAME)
+VENV = str(VENV_PATH.expanduser())
```

Comment:
I prefer activating a suitable virtual environment manually. In particular because I (and maybe others) don't use `virtualenv` but `conda`.

Reply:
I was just following the standard of all the other plugins and pelican itself. All the other ones are using virtual env and in order to keep everything consistent it would make sense to replicate those no?

Reply:
Well, not if it disrupts the preferred tooling of plugin authors and others who prefer alternate workflows. 😉

Reply:
ok. Sorry .-. I'll go ahead and undo this as well.

Reply:
So for this one. If I changed it so that it used the same thing like poetry:
```
POETRY = which("poetry") if which("poetry") else (VENV / Path("bin") / "poetry")
```
would that be ok @oulenz ? It would be the ebst of both worlds right? It uses paths when they exist, else it looks inside the environment bin. Or would you prefer I just use the same method that you had before?

Reply:
I think that would work. I have to admit, I am getting utterly confused by how all these tools operate. In this case, what I don't understand is why we'd want to use invoke to call poetry, rather than the other way around. If you do `poetry run invoke ...`, doesn't poetry then handle the environment, so we could have a cleaner `tasks.py` file? :thinking:

Reply:
Ok, I'll go ahead and do that for now =) Maybe @justinmayer  would have more idea why we use invoke for something like this? I'm still a newbie... .-.

Reply:
My task automation priorities have been to (1) be flexible and accommodate different workflows and (2) make it easy for folks to set up a Pelican development environment. Regarding the first priority, there are (as we've discussed) many ways to handle virtual environments. I prefer to manage mine manually, rather than let Poetry manage them for me. At the same time, people should also be free to let Poetry handle virtual environments on their behalf. Along similar lines, some people (like me) prefer having Poetry and Pre-commit installed globally instead of having them only live in virtual environments. Some folks don't.

Regarding the second objective, the reason I use Invoke to call Poetry here is to keep things simple for new contributors. By putting as much as possible into Invoke tasks, that means that installing Invoke becomes the primary pre-requisite, and then Invoke can handle most of the rest. So in theory, `pip install invoke && invoke setup` becomes all that's required to get going. The concept of letting Poetry manage the environment and then using `poetry run invoke …` is valid enough, but that would depend on Poetry being globally installed, and I didn't want to force users to globally install anything just in order to contribute to the project. The way I configured tasks.py and the [relevant instructions](http://docs.getpelican.com/en/latest/contribute.html#setting-up-the-development-environment) is that it all works with a minimal number of steps and regardless of whether Poetry and Pre-commit are globally installed or not.

Reply:
Thanks for the explanation, it's helpful to understand what is logically necessary and what are design decisions. I agree that focusing everything on invoke and `tasks.py` is probably the easiest to understand for new contributors.